cordapp/src/main/kotlin/net/corda/plugins/SigningOptions.kt

```diff
@@ -2,207 +2,160 @@ package net.corda.plugins
 
 import org.gradle.api.tasks.Input
 
-/** JAR sign options. */
-//Option used for ANT task "signjar"
+/** Options for ANT task "signjar". */
 open class SigningOptions {
     companion object {
-        // Defaults to resource/certificates/cordadevcakeys.jks keyStore with Corda development key
+        // Defaults to resource/certificates/cordadevcakeys.jks keystore with Corda development key
         private const val DEFAULT_ALIAS = "cordaintermediateca"
         private const val DEFAULT_STOREPASS = "cordacadevpass"
         private const val DEFAULT_STORETYPE = "JKS"
         private const val DEFAULT_KEYPASS = "cordacadevkeypass"
         const val DEFAULT_KEYSTORE = "certificates/cordadevcakeys.jks"
         const val DEFAULT_KEYSTORE_FILE = "cordadevcakeys"
         const val DEFAULT_KEYSTORE_EXTENSION = "jks"
+        const val SYSTEM_PROPERTY_PREFIX = "signing."
     }
 
-    // Methods accept null value, this allows to use e.g. 'alias System.getProperty('key')' in Gradle configuration file,
-    // instead of requiring 'alias System.getProperty('key','')'
-    @get:Input
-    var alias = ""
-
-    fun alias(value: String?) {
-        alias = value ?: ""
+    /** Option keys for ANT task. */
+    class Key {
+        companion object {
+            val JAR = "jar"
+            val ALIAS = "alias"
+            val STOREPASS = "storepass"
+            val KEYSTORE = "keystore"
+            val STORETYPE = "storetype"
+            val KEYPASS = "keypass"
+            val SIGFILE = "sigfile"
+            val SIGNEDJAR = "signedjar"
+            val VERBOSE = "verbose"
+            val STRICT = "strict"
+            val INTERNALSF = "internalsf"
+            val SECTIONSONLY = "sectionsonly"
+            val LAZY = "lazy"
+            val MAXMEMORY = "maxmemory"
+            val PRESERVELASTMODIFIED = "preservelastmodified"
+            val TSACERT = "tsaurl"
+            val TSAURL = "tsacert"
+            val TSAPROXYHOST = "tsaproxyhost"
+            val TSAPROXYPORT = "tsaproxyport"
+            val EXECUTABLE = "executable"
+            val FORCE = "force"
+            val SIGALG = "sigalg"
+            val DIGESTALG = "digestalg"
+            val TSADIGESTALG = "tsadigestalg"
+        }
     }
 
     @get:Input
-    var storepass = ""
-
-    fun storepass(value: String?) {
-        storepass = value ?: ""
-    }
+    var alias = System.getProperty(SYSTEM_PROPERTY_PREFIX + Key.ALIAS, DEFAULT_ALIAS)
```

Comment:
The fact these will resolve to null isn't a problem?

Reply:
It' ok, if not specified in plugin config it will try system property and then the hardcoded `DEFAULT_ALIAS`, or did you mean something else?

Reply:
Aha, I missed the second parameter to getProperty.